packages/client/src/game/rules/variants/sudoku.ts

```diff
@@ -163,30 +214,46 @@ export function getMaxScorePerStack(
   // Same, but sorted in ascending order.
   let bestAssignmentSorted: number[] = [];
 
-  let localSuitIndex = 0;
-
-  const curAssignment: number[] = [];
-  for (const i of eRange(curAssignment.length)) {
-    curAssignment[i] = unassigned;
-  }
-  const assigned: boolean[] = [];
-
-  while (localSuitIndex >= 0) {
-    if (curAssignment[localSuitIndex] !== unassigned) {
-      assigned[curAssignment[localSuitIndex]!] = false;
+  // This will denote a 'local' index and refers to the index in the unassignedSuits array. We will
+  // therefore iterate this over 0, ..., unassignedSuits.length - 1.
+  let localSuitIndex: FiveStackIndex = 0;
+
+  // A map (unassignedSuits) -> (index in possibleStackStarts) that denotes the current assignment
+  // of the stacks to their starting values. We use local suit indices to access into this.
+  const curAssignment: Tuple<FiveStackIndex | undefined, 5> = [
+    undefined,
+    undefined,
+    undefined,
+    undefined,
+    undefined,
+  ];
+
+  // A map (index of stackStart) -> bool, denoting wether some suit is currently assigned to this
+  // stack start.
+  const assigned: Tuple<boolean, 5> = [false, false, false, false, false];
+
+  while (true) {
```

Comment:
I'm unsure on how to fix this linter error.
the localSuitIndex is incremented/decremented during the loop, and we want to stop the loop once it would be decremented to -1 (see line 349).